Patch-Source: https://gitlab.gnome.org/GNOME/polari/-/commit/d7946c7fe39f112cd3f751bb95b170446022980d
---
From d7946c7fe39f112cd3f751bb95b170446022980d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Sat, 19 Jul 2025 18:26:55 +0200
Subject: [PATCH] Port to gjs-1.85/girepository-2.0

gjs ported from the stand-alone gobject-introspection-1.0 to the
new girepository-2.0 library bundled with glib.

Bump the requirement for gjs/glib and adjust to the (fairly minor)
changes.

Part-of: <https://gitlab.gnome.org/GNOME/polari/-/merge_requests/356>
---
 meson.build  | 9 ++++++---
 src/polari.c | 8 +++++---
 2 files changed, 11 insertions(+), 6 deletions(-)

diff --git a/meson.build b/meson.build
index 88b5859b..e126e55c 100644
--- a/meson.build
+++ b/meson.build
@@ -38,11 +38,14 @@ json_glib_validate = find_program('json-glib-validate', required: false)
 appstreamcli = find_program('appstreamcli', required: false)
 check_version = find_program('build-aux/meson/check-version.py')
 
-gio = dependency('gio-2.0', version: '>= 2.43.4')
+gio_req = '>= 2.85.1'
+gjs_req = '>= 1.85.1'
+
+gio = dependency('gio-2.0', version: gio_req)
 telepathy_glib = dependency('telepathy-glib')
 tracker = dependency('tracker-sparql-3.0')
-girepository = dependency('gobject-introspection-1.0')
-gjs = dependency('gjs-1.0', version: '>= 1.73.1')
+girepository = dependency('girepository-2.0', version: gio_req)
+gjs = dependency('gjs-1.0', version: gjs_req)
 
 conf = configuration_data()
 
diff --git a/src/polari.c b/src/polari.c
index 48f5bde8..33fc4786 100644
--- a/src/polari.c
+++ b/src/polari.c
@@ -3,7 +3,7 @@
  *
  * SPDX-License-Identifier: GPL-2.0-or-later
 */
-#include <girepository.h>
+#include <girepository/girepository.h>
 #include <gjs/gjs.h>
 
 #include "config.h"
@@ -67,6 +67,7 @@ main (int argc, char *argv[])
   const char *search_path[] = { "resource:///org/gnome/Polari/js", NULL };
   g_autoptr (GOptionContext) option_context = NULL;
   g_autoptr (GError) error = NULL;
+  g_autoptr (GIRepository) repo = NULL;
   g_autoptr (GjsContext) context = NULL;
   g_auto (GStrv) js_argv = NULL;
   GjsProfiler *profiler = NULL;
@@ -86,8 +87,9 @@ main (int argc, char *argv[])
   g_set_application_name ("Polari");
 #endif
 
-  g_irepository_prepend_search_path (PKGLIBDIR "/girepository-1.0");
-  g_irepository_prepend_library_path (PKGLIBDIR);
+  repo = gi_repository_dup_default ();
+  gi_repository_prepend_search_path (repo, PKGLIBDIR "/girepository-1.0");
+  gi_repository_prepend_library_path (repo, PKGLIBDIR);
 
   context = g_object_new (GJS_TYPE_CONTEXT,
                           "search-path", search_path,
-- 
GitLab

